src/components/input.js

```diff
@@ -236,7 +236,7 @@ export class AutocompleteTextInput extends Component {
       },
       renderSuggestion: v => v,
       renderInputComponent: inputProps => {
-        return textInput({ ...props, ...inputProps, type: 'search' })
+        return textInput({ ...props, ...inputProps, style: containsIcon ? { borderRadius: '4px 0px 0px 4px', borderRight: 'white' } : undefined, type: 'search' })
```

Comment:
I'm not sure we're confident enough in the design of this to elevate it to the shared component. Instead, let's tweak this just enough to enable arbitrary style overrides, and then send them in from the other side.